Dockerfile

```diff
@@ -4,66 +4,94 @@ ARG RUBY_VERSION=3.4.4
 
 FROM docker.io/library/ruby:$RUBY_VERSION-slim AS base
 
-ENV RACK_ENV=production
-ENV HANAMI_ENV=production
-ENV HANAMI_SERVE_ASSETS=true
-ENV BUNDLE_DEPLOYMENT=1
 ENV BUNDLE_PATH=/usr/local/bundle
-ENV BUNDLE_WITHOUT="development:quality:test:tools"
-
 WORKDIR /app
 
+# Combined repository setup and package installation
 RUN <<STEPS
-  apt-get update -qq \
-  && apt-get install --no-install-recommends -y \
-  chromium \
-  curl \
-  imagemagick \
-  libjemalloc2 \
-  nodejs \
-  npm \
-  postgresql-client \
-  tmux \
-  && rm -rf /var/lib/apt/lists /var/cache/apt/archives
+  apt-get update -qq
+  apt-get install -y --no-install-recommends curl ca-certificates gnupg2 lsb-release
+  # Setup PostgreSQL repository
+  curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor -o /usr/share/keyrings/postgresql-keyring.gpg
+  echo "deb [signed-by=/usr/share/keyrings/postgresql-keyring.gpg] https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+  # Setup Node.js repository and install runtime dependencies
+  curl -fsSL https://deb.nodesource.com/setup_23.x | bash -
+  apt-get update -qq
+  apt-get install -y --no-install-recommends \
+      chromium \
+      imagemagick \
+      libjemalloc2 \
+      tmux \
+      postgresql-client-17 \
+      nodejs
+  rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
 STEPS
 
+# Build stage for installing dependencies
 FROM base AS build
 
+# Install build dependencies
 RUN <<STEPS
-  apt-get update -qq \
-  && apt-get install --no-install-recommends -y build-essential \
-  libpq-dev \
-  libyaml-dev \
-  pkg-config \
-  && rm -rf /var/lib/apt/lists /var/cache/apt/archives
+  apt-get update -qq
+  apt-get install -y --no-install-recommends \
+      build-essential \
+      libpq-dev \
+      libyaml-dev \
+      pkg-config
+  rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
```

Comment:
:thought_balloon: Why the use of splats here? Is deleting `lists` and `archives` not enough?

Reply:
Sounds good. Let's remove for now to keep the diff lower. We can always refactor more in the future if a specific problem arises.

Reply:
The only other change to not here is since we are using the `STEPS` heredoc this sill be run as one statment and cached the same way in docker so we don't need to use the `\` syntax to get multiline. that was the real motivation behind this change.

Reply:
Fair point. Yeah, that'll be a nice correction.